### PR TITLE
PHPStan level 5 + upgrade to 0.11.8

### PR DESCRIPTION
**Pull request by mhlavac**
* Updated phpstan to version 0.11.8
* Bumped phpstan level to 5
* Fixed clearstatcache location being used as 1st parameter instead
  as 2nd parameter
* Fixed config fallback incorrect phpdoc

### CODE REVIEW

**Comment by frankdejonge** [comment]
The clearstatscache call was already fixed in master.

**Comment by mhlavac** [comment]
@frankdejonge great, I just saw you merged the other pull request. I'll rebase this on master in a minute. I was surprised that phpstan found the issue as well.

Would you mind if I PR local dev set up, so php-cs-fixer, phpstan and other tools can be run via single command in local? I think it will ease the contribution effort

**Comment by frankdejonge** [comment]
Can you set that up as a composer script?

**Comment by mhlavac** [comment]
Yes, I think that should be possible

**Comment by mhlavac** [comment]
Rebased and updated the PR to reflect what it changes

**Comment by frankdejonge** [comment]
I'm about to do a release, so this will at the earliest go into the next release.

**Comment by mhlavac** [comment]
I'll open separate PR for adding the scripts as composer scripts

File: composer.json [hunk 1]
```diff
@@ -19,7 +19,8 @@
     },
     "require-dev": {
         "phpspec/phpspec": "^3.4",
-        "phpunit/phpunit": "^5.7.10"
+        "phpunit/phpunit": "^5.7.10",
+        "phpstan/phpstan": "^0.11.8"
```
Comment (frankdejonge) [inline composer.json]: adding this dev dependency causes unwanted failures in the test suite, please check the travis output.

**Comment by mhlavac** [comment]
@frankdejonge fixed, I had hard times to get travis-ci script to work, so instead I implemented short php script to download phpstan shim that does the job. Is it reasonable?

**Pull request merged**